src/SourceBuild/content/repo-projects/Directory.Build.targets

```diff
@@ -462,6 +461,15 @@
   <Target Name="CleanupRepo"
           AfterTargets="RemoveBuiltPackagesFromCache"
           Condition="'$(CleanWhileBuilding)' == 'true' and Exists('$(ProjectDirectory)artifacts')">
+
+    <!--
+      Some repositories (WinForms) use source generators that open files manually and keep file handles open on the compiler server (CsWin32).
+      These source generators are written incorrectly (a source generator should never do IO itself),
+      but the required features to read in binary files as input to a source generator don't exist.
+      To work around these source generators, shut down the compiler server so we can delete the obj directories.
+    -->
+    <Exec Command="$(DotnetTool) build-server shutdown --vbcscompiler" />
```

Comment:
Will this have any effect on parallel builds?

Reply:
Roslyn is already resilient to the compilation server disappearing. Worst case, it slows down the parallel builds because it kills the server in the middle of a compilation.

I'll move this into a custom target in the WinForms repo-project so it's only done when necessary.

Reply:
Actually, we already shut down all of the build servers at the start of each build (see line 285), so this is no worse for parallelism than the existing infrastructure.

Reply:
Ahh, and this is only required for clean while building, got it.